traitsui/testing/tester/wx/helpers.py

```diff
@@ -11,6 +11,63 @@
 
 import wx
 
+from traitsui.testing.tester.exceptions import Disabled
+from traitsui.wx.key_event_to_name import key_map as _KEY_MAP
+
+
+def apply_modifiers(key_event, modifiers):
+    possible_modifiers = ["Alt", "Ctrl", "Meta", "Shift"]
+    modifier_map = {mod: (mod in modifiers) for mod in possible_modifiers}
+    key_event.SetAltDown(modifier_map["Alt"])
+    key_event.SetControlDown(modifier_map["Ctrl"])
+    key_event.SetMetaDown(modifier_map["Meta"])
+    key_event.SetShiftDown(modifier_map["Shift"])
+
+
+def key_click(widget, key, delay=0):
+    """ Performs a key click of the given key on the given widget after
+    a delay.
+
+    Parameters
+    ----------
+    widget : wxObject
+        The wx Object to be clicked. Should be wx.TextCtrl instance
+    key : str
+        Standardized (pyface) name for a keyboard event.
+        e.g. "Enter", "Tab", "Space", "0", "1", "A", ...
+    delay : int 
+        Time delay (in ms) in which the key click will be performed.
+    """
+    if "-" in key:
+        *modifiers, key = key.split("-")
+    else:
+        modifiers = []
+
+    mapping = {name: event for event, name in _KEY_MAP.items()}
+    if key not in mapping:
+        if len(key) == 1:
+            try:
+                KEY = ord(key)
+            except:
+                raise ValueError(
+                    "Unknown key {!r}. Expected one of these: {!r}, or a unicode character".format(
+                        key, sorted(mapping)
+                    ))
+            else:
+                wx.MilliSleep(delay)
+                key_event = wx.KeyEvent(wx.wxEVT_CHAR)
+                apply_modifiers(key_event, modifiers)
+                key_event.SetUnicodeKey(KEY)
+                widget.EmulateKeyPress(key_event)
+
+    else:
+        wx.MilliSleep(delay)
+        KEY = mapping[key]
+        key_event = wx.KeyEvent(wx.wxEVT_CHAR)
+        apply_modifiers(key_event, modifiers)
+        key_event.SetKeyCode(mapping[key])
+        widget.EmulateKeyPress(key_event)
```

Comment:
I attempted to add functionality for handling modifiers.  However, this code is currently not working.  It may be due to drawbacks with the EmulateKeyPress method (https://discuss.wxpython.org/t/wx-textctrl-and-emulatekeypress/28300/3), it may not...
Currently, the code allows for simulated key clicks/sequences for standard characters, and pyface key names (e.g. "Backspace").
However, trying to perform key clicks with things like "Shift-A", or "Ctrl-C" currently do not work.